ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
+    expect(getRunningCost()).toEqual('$0.20 per hour');
+    expect(getPausedCost()).toEqual('< $0.01 per hour');
+
+    // Change the machine to n1-standard-8 and bump the storage to 300GB.
+    await pickMainCpu(container, 8);
+    await pickMainRam(container, 30);
+    await pickDetachableDiskSize(300);
+    expect(getRunningCost()).toEqual('$0.40 per hour');
+    expect(getPausedCost()).toEqual('$0.02 per hour');
+
+    await pickPresets(container, runtimePresets.generalAnalysis.displayName);
+    expect(getRunningCost()).toEqual('$0.20 per hour');
+    expect(getPausedCost()).toEqual('< $0.01 per hour');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    expect(getRunningCost()).toEqual('$0.73 per hour');
+    expect(getPausedCost()).toEqual('$0.02 per hour');
+
+    // Bump up all the worker values to increase the price on everything.
+    await pickNumWorkers(4);
+    await pickNumPreemptibleWorkers(4);
+    await pickWorkerCpu(container, 8);
+    await pickWorkerRam(container, 30);
+    await pickWorkerDiskSize(300);
+    expect(getRunningCost()).toEqual('$2.88 per hour');
+    expect(getPausedCost()).toEqual('$0.14 per hour');
+  });
+
+  it('should update the cost estimator when master machine changes', async () => {
+    setCurrentRuntime({
+      ...runtimeApiStub.runtime,
+      status: RuntimeStatus.RUNNING,
+      configurationType: RuntimeConfigurationType.USER_OVERRIDE,
+      gceConfig: null,
+      gceWithPdConfig: null,
+      dataprocConfig: {
+        masterMachineType: 'n1-standard-4',
+        masterDiskSize: 1000,
+        numberOfWorkers: 2,
+        numberOfPreemptibleWorkers: 0,
+        workerMachineType: 'n1-standard-4',
+        workerDiskSize: DATAPROC_MIN_DISK_SIZE_GB,
+      },
+    });
+
+    const { container } = await component();
+
+    // with Master disk size: 1000
+    expect(screen.getByText('Cost when running')).toBeTruthy();
+    expect(screen.getByText('Cost when paused')).toBeTruthy();
+
+    expect(getRunningCost()).toEqual('$0.77 per hour');
+    expect(getPausedCost()).toEqual('$0.07 per hour');
+
+    // Change the Master disk size or master size to 150
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+
+    expect(screen.getByText('Cost when running')).toBeTruthy();
+    expect(screen.getByText('Cost when paused')).toBeTruthy();
+
+    expect(getRunningCost()).toEqual('$0.73 per hour');
+    expect(getPausedCost()).toEqual('$0.02 per hour');
+    // Switch to n1-highmem-4, double disk size.
+    await pickMainRam(container, 26);
+    await pickStandardDiskSize(2000);
+    expect(getRunningCost()).toEqual('$0.87 per hour');
+    expect(getPausedCost()).toEqual('$0.13 per hour');
+  });
+
+  it('should prevent runtime update when disk size is invalid', async () => {
+    const { container } = await component();
+
+    const getNextButton = () => screen.getByRole('button', { name: 'Next' });
+
+    await pickDetachableDiskSize(49);
```

Comment:
We can remove the detachable-disks part of this test, because it's covered in the next test.

Reply:
I want to confirm if i understand the change correct, we can discuss this in mob

Reply:
Moved to another test